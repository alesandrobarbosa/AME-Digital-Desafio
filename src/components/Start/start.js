import React from 'react';
import { Link } from 'react-router-dom';
import './start.scss';


const Start = () => {
    let logoImg = require('../../view/img/star-wars-logo.png');
    
    return (
        <div className="start">
            <img className="start__logo" src={logoImg} alt="logo" />
            <Link className="start__play" to="/game">START</Link>
        </div>
    )
}


export default Start;