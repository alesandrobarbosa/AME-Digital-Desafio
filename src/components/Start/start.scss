@import '../../view/styles/styles.scss';

.start {
    margin: auto;
    display: flex;
    flex-flow: column;
    &__play {
        background: $btnBgColor;
        border-radius: 10px;
        border-bottom: 6px solid #cccccc;
        padding: 20px 50px;
        margin: 30px auto;
        font-family: $fontStyle;
        text-decoration: none;
        color: $btnTxtColor;
        text-transform: uppercase;
        font-size: 1.2em;
        &:active {
            border: none;
            min-height: 25px;
        }
    }
    &__logo {
        width: 100%;
    }
}
