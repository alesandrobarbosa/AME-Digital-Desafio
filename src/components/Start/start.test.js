import React from 'react';
import renderer from 'react-test-renderer';

import { MemoryRouter } from 'react-router-dom';
import Start from './start';

describe('#START COMPONENT ==>', () => {
    it('Start should renders without crashing', () => {
        const component = renderer.create(
            <MemoryRouter>
                <Start />
            </MemoryRouter>
        ).toJSON();
    
        expect(component).toMatchSnapshot();
    });

})

