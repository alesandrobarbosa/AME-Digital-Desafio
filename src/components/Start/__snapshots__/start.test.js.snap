// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#START COMPONENT ==> Start should renders without crashing 1`] = `
<div
  className="start"
>
  <span
    className="start__first"
  >
    Desafio AME Digital
  </span>
  <img
    alt="logo"
    className="start__logo"
    src="star-wars.png"
  />
  <span
    className="start__second"
  >
    por 
    <strong>
      Alesandro Barbosa
    </strong>
  </span>
  <a
    className="start__play"
    href="/game"
    onClick={[Function]}
  >
    START
  </a>
</div>
`;
